cpp/src/arrow/compute/function.cc

```diff
@@ -167,6 +179,109 @@ const Kernel* DispatchExactImpl(const Function* func,
   return nullptr;
 }
 
+struct FunctionExecutorImpl : public FunctionExecutor {
+  FunctionExecutorImpl(std::vector<TypeHolder> in_types, const Kernel* kernel,
+                       std::unique_ptr<detail::KernelExecutor> executor,
+                       const Function& func)
+      : in_types(std::move(in_types)),
+        kernel(kernel),
+        kernel_ctx(default_exec_context(), kernel),
```

Comment:
Is this useful? It will be overriden in `Init` anyway.

Reply:
`Init` may be called zero or more times. For the zero-times case, this sets the default.

Reply:
Oops, sorry, I keep forgetting.